src/interface/compute.jl

```diff
@@ -187,7 +187,41 @@ function (ctx::SuitableRep)(ex)
         return data_rep(ex.tns.val)
     elseif ex.kind === mapjoin
         ##Assumes concordant mapjoin arguments, probably okay
-        return map_rep(ex.op.val, map(ctx, ex.args)...)
+        output_idxs = getfields(ex, ctx.bindings)
+        args_idxs = map(getfields, ex.args, ctx.bindings)
+        args_reps = map(ctx, ex.args)
+        idx_to_args_styles = Dict(i => [] for i in union(args_idxs...))
+        for i in eachindex(ex.args)
+            idxs = args_idxs[i]
+            arg_rep =args_reps[i]
+            for idx in idxs
+                push!(idx_to_args_styles[idx], map_rep_style(arg_rep))
+                arg_rep = map_rep_child(arg_rep)
+            end
+        end
+        output_styles = []
+        for idx in output_idxs
+            args_styles = idx_to_args_styles[idx]
+            if length(args_styles) == 1
+                push!(output_styles, args_styles[1])
+            else
+                push!(output_styles, result_style(args_styles...))
+            end
+        end
+
+        output_def = ElementData(ex.op.val(map(default, args_reps)...), combine_eltypes(ex.op.val, (args_reps...,)))
+        for style in reverse(output_styles)
+            if typeof(style) == Finch.MapRepDenseStyle
+                output_def = DenseData(output_def)
+            elseif typeof(style) == Finch.MapRepSparseStyle
+                output_def = SparseData(output_def)
+            elseif typeof(style) == Finch.MapRepRepeatStyle
+                output_def = RepeatData(output_def)
+            elseif typeof(style) == Finch.MapRepExtrudeStyle
+                output_def = ExtrudeData(output_def)
+            end
+        end
+        return output_def
```

Comment:
I think this would probably be more cleanly represented with extrude_rep, but I need to investigate a little more here. Does your test case exemplify the problem you faced?

Reply:
Yeah, the test case should show this problem. I think new functionality is needed because the format selection is dependent on the named indices, i.e. in order to differentiate between A[i,j]*B[j,k] and A[i,j]*B[i,j]. However, the way I've written it here doesn't necessarily fit with the broader code style. I could put this logic in traits.jl potentially. I'd just need to include index information in an updated map_rep interface.